gateways/cardconnect/types.go

```diff
@@ -0,0 +1,90 @@
+package cardconnect
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type CardConnectClient struct {
+	username   string
+	password   string
+	merchantID string
+	httpClient *http.Client
+	URL        string
+}
+
+func UnmarshalRequest(data []byte) (Request, error) {
+	var r Request
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *Request) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+type Request struct {
```

Comment:
I don't see billing address, just confirming that this is intended to be missing from the auth request

Reply:
https://developer.cardpointe.com/cardconnect-api#authorization-request
It has names, address, address2, city, region, country, phone, and email fields. Do I need to fill them too?

Reply:
Yeah, merchant's will get better auth rates the more information we provide.